File: CMD/app/src/main/java/com/tmdhoon/cmd/Api/ServerApi.java
package com.tmdhoon.cmd.Api;

import android.text.Editable;

import com.tmdhoon.cmd.Request.LoginRequest;
import com.tmdhoon.cmd.Request.RegisterRequest;
import com.tmdhoon.cmd.Response.LoginResponse;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface ServerApi {

    @POST("signup/{secretKey}")
    Call<Void> register(
            @Path("secretKey") String secretKey,
            @Body RegisterRequest registerRequest
    );

    @POST("signin")
    Call<LoginResponse> login(
            @Body LoginRequest loginRequest
    );


}

File: CMD/app/src/main/java/com/example/cmd/Kotlin/Api/ServerApi.kt
package com.example.cmd.Kotlin.Api

import com.example.cmd.Request.SignupRequest
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.Path
import javax.crypto.SecretKey

interface ServerApi {

    @POST("signup/{secretKey}")
    fun signUp(
        @Path("secretKey") secretKey: String,
        @Body signupRequest: SignupRequest
    ) : Call<Void>



}